tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,338 @@ use strict;
     assert_screen 'yast2_samba-server_new-share_create';
-    wait_screen_change { send_key 'alt-o' };
+    send_key 'alt-o';
     assert_screen 'yast2_samba-server_new-share-path';
-    wait_screen_change { send_key 'alt-y' };
+    send_key 'alt-y';                           # confirm new path
 
     # back to samba configuration and make some changes to share directories
-    wait_screen_change { send_key 'alt-w' };    # allow users to share directories
+    assert_screen 'yast2_samba-added_html_share';
+    send_key 'alt-w';                           # allow users to share directories
     wait_screen_change { send_key 'alt-g' };    # allow guest access
-    wait_screen_change { send_key 'alt-m' };
-    for (1 .. 10) { send_key 'down'; }          # Maximum number of shares
+    send_key 'alt-m';
+    type_string($samba_directives{usershare_max_shares} . "\n");    # Maximum number of shares
 
-    # switch to identity configuration
+    ## switch to identity configuration
     send_key 'alt-d';
     assert_screen 'yast2_samba-server_identity';
 
+    ## domain controller in sle15
+    # select "Not a Domain Controller"
+    if ((is_sle('>=15')) || (is_leap('>=15.0')) || is_tumbleweed) {
+        wait_screen_change { send_key 'alt-a' };
+        send_key 'ret';
+        # check "Not a DC" is select
+        assert_screen 'yast2_samba-server_not-a-dc_selected';
+    }
+
     # use wins server support and check NetBIOS hostname Advanced settings
     wait_screen_change { send_key 'alt-i' };
     wait_screen_change { send_key 'alt-e' };
-    type_string 'QA-Samba';
+    type_string($samba_directives{netbios_name});
     send_key 'alt-v';
     assert_screen 'yast2_samba-server_identity_netbios_advanced_expert';
     send_key 'ret';
     assert_screen 'yast2_samba-server_netbios_name_change_warning';
     wait_screen_change { send_key 'alt-o' };
-
     # change logon drive to C:
     send_key_until_needlematch 'yast2_samba-server_netbios_logon-drive', 'down';
     wait_screen_change { send_key 'ret' };
-    wait_screen_change { send_key 'alt-l' };
-    for (1 .. 5) { send_key 'backspace'; }
-    type_string 'C:';
-    wait_screen_change { send_key 'alt-o' };
+    for (1 .. 2) { send_key 'backspace'; }
+    type_string($samba_directives{logon_drive});
     wait_screen_change { send_key 'alt-o' };
+    send_key 'alt-o';
 
-    # swith to Trusted Domains
-    # add a trusted domain
+    ## swith to Trusted Domains
+    # add a trusted domain, wait while all data is properly loaded
+    wait_still_screen(2);
     send_key 'alt-t';
     assert_screen 'yast2_samba-server_trusted-domains';
     wait_screen_change { send_key 'alt-a' };
-    type_string 'suse.de';
+    type_string('suse.de');
     wait_screen_change { send_key 'alt-p' };
-    type_string 'testing';
+    type_string('testing');
     send_key 'alt-o';
     assert_screen 'yast2_samba-server_trusted-domains_error';
-    wait_screen_change { send_key 'alt-o' };
-
+    send_key 'alt-o';
```

Comment:
It would be nice to assert key press here too, two key presses in a row can get pretty unstable =(